core/dom/element.js

```diff
@@ -2148,6 +2148,17 @@ CKEDITOR.dom.element.clearMarkers = function( database, element, removeFromDatab
 			} else {
 				nativeElement[ nativeElement[ eventName ] ? eventName : handlerName ]( evt );
 			}
+		},
+
+		/**
+		 * Checks if element is not attached to DOM.
+		 *
+		 * @since 4.13.0
+		 * @returns {Boolean} Whether element is detached from DOM.
+		 */
+		isDetached: function() {
+			return !this.getDocument().getDocumentElement().contains( this ) ||
```

Comment:
This condition looks a little bit unreadable. I'd split it into two parts (variables):

* `isInDocument` → `this.getDocument().getDocumentElement().contains( this )`,
* `isDocumentAttached` → `( ( !CKEDITOR.env.ie || CKEDITOR.env.version > 8 ) && this.getDocument().$.defaultView )`.

I'm also wondering about the second case in IE8. I'm feeling that we can polyfill it by something like `this.getDocument().$.parentWindow.frameElement` and then check if `frameElement` is inside its own document. However I'm not sure if it's really worth the trouble.

Reply:
I feel like mentioned polyfill is not suitable here or I just don't how it supposed to work :(
In original solution what I understand we cover case where the entire document is detached with method like: `document.implementation.createHTMLDocument()`. In such a case, there is no related `window` object and we get null. I couldn't find any possibility of how to create a detached document on IE8. That's why I'm not sure how checking the `frameElement` and its presence in own document might replace it.
Could you provide more details about this polyfill? For now, I just ignore this check on IE8 and remain current functionality.

Reply:
I made early returns to simplify this function.